azurelinuxagent/ga/remoteaccess.py

```diff
@@ -48,108 +47,107 @@ def get_remote_access_handler(protocol):
+                    except Exception as e:
+                        logger.error("Error removing remote access user '{0}' - {1}", user, ustr(e))
+
+    @staticmethod
+    def _is_jit_user(comment):
         return comment == REMOTE_ACCESS_ACCOUNT_COMMENT
 
-    def add_user(self, username, encrypted_password, account_expiration):
+    def _add_user(self, username, encrypted_password, account_expiration):
+        user_added = False
+
         try:
             expiration_date = (account_expiration + timedelta(days=1)).strftime(DATE_FORMAT)
-            logger.verbose("Adding user {0} with expiration date {1}".format(username, expiration_date))
-            self.os_util.useradd(username, expiration_date, REMOTE_ACCESS_ACCOUNT_COMMENT)
-        except Exception as e:
-            raise RemoteAccessError("Error adding user {0}. {1}".format(username, ustr(e)))
-        try:
-            prv_key = os.path.join(conf.get_lib_dir(), TRANSPORT_PRIVATE_CERT)
-            pwd = self.cryptUtil.decrypt_secret(encrypted_password, prv_key)
-            self.os_util.chpasswd(username, pwd, conf.get_password_cryptid(), conf.get_password_crypt_salt_len())
-            self.os_util.conf_sudoer(username)
-            logger.info("User '{0}' added successfully with expiration in {1}".format(username, expiration_date))
-        except Exception as e:
-            error = "Error adding user {0}. {1} ".format(username, str(e))
-            try:
-                self.handle_failed_create(username)
-                error += "cleanup successful"
-            except RemoteAccessError as rae:
-                error += "and error cleaning up {0}".format(str(rae))
-            raise RemoteAccessError("Error adding user {0} cleanup successful".format(username), ustr(e))
-
-    def handle_failed_create(self, username):
-        try:
-            self.delete_user(username)
-        except Exception as e:
-            raise RemoteAccessError("Failed to clean up after account creation for {0}.".format(username), e)
+            logger.info("Adding remote access user '{0}' with expiration date {1}", username, expiration_date)
+            self._os_util.useradd(username, expiration_date, REMOTE_ACCESS_ACCOUNT_COMMENT)
+            user_added = True
 
-    def remove_user(self, username):
-        try:
-            self.delete_user(username)
-        except Exception as e:
-            raise RemoteAccessError("Failed to delete user {0}".format(username), e)
+            logger.info("Adding remote access user '{0}' to sudoers", username)
+            prv_key = os.path.join(conf.get_lib_dir(), TRANSPORT_PRIVATE_CERT)
+            pwd = self._cryptUtil.decrypt_secret(encrypted_password, prv_key)
+            self._os_util.chpasswd(username, pwd, conf.get_password_cryptid(), conf.get_password_crypt_salt_len())
+            self._os_util.conf_sudoer(username)
+        except Exception:
+            if user_added:
+                self._remove_user(username)
+            raise
```

Comment:
This except: block (lines 145-148) looks like it would be more natural as a finally: block.